src/services/SubmitTxMonitor.ts

```diff
@@ -1,35 +1,42 @@
-import {injectable} from 'inversify';
-import fs from 'fs';
+import {inject, injectable} from 'inversify';
 
 import {ChainsIds} from '../types/ChainsIds';
-
-type SubmitMonitor = {
-  dataTimestamp: number;
-  txHash: string;
-};
+import {MappingRepository} from '../repositories/MappingRepository';
+import {SubmitMonitor} from '../types/SubmitMonitor';
 
 @injectable()
 export class SubmitTxMonitor {
-  private filePattern = '/tmp/submit-$chainId.json';
+  @inject(MappingRepository) mappingRepository!: MappingRepository;
 
-  monitorFile = (chainId: ChainsIds): string => {
-    return this.filePattern.replace('$chainId', chainId);
-  };
+  monitorKey(chainId: ChainsIds): string {
+    return `SUBMIT_TX_MONITOR${chainId}`;
+  }
 
-  saveTx = (chainId: ChainsIds, dataTimestamp: number, txHash: string): void => {
+  async saveTx(chainId: ChainsIds, dataTimestamp: number, txHash: string): Promise<void> {
     const submitMonitor: SubmitMonitor = {dataTimestamp, txHash};
-    const file = this.monitorFile(chainId);
-    fs.writeFileSync(file, JSON.stringify(submitMonitor), 'utf-8');
-  };
+    const key = this.monitorKey(chainId);
+    await this.mappingRepository.set(key, JSON.stringify(submitMonitor));
+  }
 
-  wasDataSubmitted = (chainId: ChainsIds, dataTimestamp: number): boolean => {
-    const file = this.monitorFile(chainId);
+  async wasDataSubmitted(chainId: ChainsIds, dataTimestamp: number): Promise<boolean> {
```

Comment:
ideally a service should only have one public service. looks like this should be decomposed to its own service.

Reply:
i'd name this something like `SubmitMonitorTxChecker` or something like that

Reply:
Often there is a need to have few public methods, that are working based on some common core code, in this case `monitorKey()`. All of this methods fits in one logical entity quite good, you can easily see whole picture of functionality without need of review more files. How would you call such an object with few public methods? Repository would be better name? I can move it to `repository/` if you think it fits better there.

If I would split this code, we will end up with few small (few lines) services, that you have to include in other files to get full functionality. So I think in such a cases, it is better to have few public methods.

I renamed to `SubmitMonitorTxChecker`.